crates/tests/src/rpc/read.rs

```diff
@@ -0,0 +1,1207 @@
+                                poseidon_builtin_applications: None,
+                                ec_op_builtin_applications: None,
+                                ecdsa_builtin_applications: None,
+                                bitwise_builtin_applications: None,
+                                keccak_builtin_applications: None,
+                                segment_arena_builtin: None,
+                            },
+                            data_resources: DataResources {
+                                data_availability: DataAvailabilityResources { l1_gas: 0, l1_data_gas: 0 },
+                            },
+                        },
+                        execution_result: ExecutionResult::Succeeded,
+                    })
+                },
+            }],
+        });
+        assert_eq!(block, expected_block);
+    }
+
+    /*
+       curl --location 'https://free-rpc.nethermind.io/sepolia-juno/' \
+       --header 'Content-Type: application/json' \
+       --data '{
+           "jsonrpc": "2.0",
+           "method": "starknet_getBlockWithTxHashes",
+           "params": {
+               "block_id": "latest"
+           },
+           "id": 1
+       }'
+    */
+    #[tokio::test]
+    async fn test_get_block_txn_with_tx_hashes_works() {
+        let madara = get_shared_state().await;
+        let json_client = JsonRpcClient::new(HttpTransport::new(madara.rpc_url.clone()));
+        let block = { json_client.get_block_with_tx_hashes(BlockId::Number(2)).await.unwrap() };
+
+        let expected_block = MaybePendingBlockWithTxHashes::Block(BlockWithTxHashes {
+            status: BlockStatus::AcceptedOnL2,
+            block_hash: Felt::from_hex_unchecked("0x7a906dfd1ff77a121b8048e6f750cda9e949d341c4487d4c6a449f183f0e61d"),
+            parent_hash: Felt::from_hex_unchecked("0x78b67b11f8c23850041e11fb0f3b39db0bcb2c99d756d5a81321d1b483d79f6"),
+            block_number: 2,
+            new_root: Felt::from_hex_unchecked("0xe005205a1327f3dff98074e528f7b96f30e0624a1dfcf571bdc81948d150a0"),
+            timestamp: 1700475581,
+            sequencer_address: Felt::from_hex_unchecked(
+                "0x1176a1bd84444c89232ec27754698e5d2e7e1a7f1539f12027f28b23ec9f3d8",
+            ),
+            l1_gas_price: ResourcePrice {
+                price_in_fri: Felt::from_hex_unchecked("0x0"),
+                price_in_wei: Felt::from_hex_unchecked("0x3b9ad016"),
+            },
+            l1_data_gas_price: ResourcePrice {
+                price_in_fri: Felt::from_hex_unchecked("0x1"),
+                price_in_wei: Felt::from_hex_unchecked("0x1"),
+            },
+            l1_da_mode: L1DataAvailabilityMode::Calldata,
+            starknet_version: "0.12.3".to_string(),
+            transactions: vec![Felt::from_hex_unchecked(
+                "0x701d9adb9c60bc2fd837fe3989e15aeba4be1a6e72bb6f61ffe35a42866c772",
+            )],
+        });
+        assert_eq!(block, expected_block);
+    }
+
+    /*
```

Comment:
I would rather comment doc using `///` since we are doing it this way all over the codebase:

```rust
/// Sends a request to fetch the latest block with transactions.
/// 
/// Example curl command:
/// 
/// ```bash
/// curl --location 'https://free-rpc.nethermind.io/sepolia-juno/' \
/// --header 'Content-Type: application/json' \
/// --data '{
///     "jsonrpc": "2.0",
///     "method": "starknet_getBlockWithTxs",
///     "params": {
///         "block_id": "latest"
///     },
///     "id": 1
/// }'
/// ```
```

Reply:
Also I would avoid referencing `https://free-rpc.nethermind.io/sepolia-juno/` and replace it with a madara public rpc (we have this one here https://rpc.kasar.io/deoxys)

Reply:
so lets rely on nethermind for now

Reply:
documentation updated